packages/syntax-core/src/getZIndex.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * [getZIndex](?path=/docs/ZIndex--docs) is a utility function that returns a z-index value based on the layer and local layer.
+ *
+ * Usage:
+ * ```
+ * <Navbar zIndex={getZIndex("sticky")} />
+ * <Modal zIndex={getZIndex("modal")} />
+ * ```
+ */
+export default function getZIndex(
+  /**
+   * The layer to get the z-index for.
+   *
+   * * `base`: 0
+   * * `menu`: 10
+   * * `sticky`: 20
+   * * `fixed`: 30
+   * * `modal`: 40
+   * * `popup`: 50
+   *
+   * @defaultValue `base`
+   */
+  layer: "base" | "menu" | "sticky" | "fixed" | "modal" | "popup" = "base",
+  /**
+   * getZIndex recognizes that there are occasional needs to have multiple things stacked in the same layer.
+   * For example, two modals existing in the same layer. In these occasions, you can pass a localLayer to the function to manually tweak the z-index within each layer.
+   *
+   * @defaultValue `0`
+   */
+  localLayer: -3 | -2 | -1 | 0 | 1 | 2 | 3 = 0,
+): number {
+  const layerIndex = {
```

Comment:
How well do these particular layers correspond to how we are using z-indexes in the app right now? I think it would be helpful to audit where we currently need z-indexes and for what types of components.

Reply:
Keep in mind that some z-indexes might not even be explicitly set in our code. For example, it seems that the MUI Dialog component has a z-index of 1300 (see [previous discussion](https://github.com/Cambly/Cambly-Frontend/pull/11895#discussion_r1750587506)), and it appears we are still using that in some places in our code.